services/server/server.go

```diff
@@ -572,6 +575,14 @@ func (s *Server) ChangeSettings(ctx context.Context, req *serverpb.ChangeSetting
 			return status.Error(codes.InvalidArgument, e.Error())
 		}
 
+		if req.DisableAlerting {
+			// Deletes the alert rules files in case of disabling
+			s.rulesService.RemoveVMAlertRulesFiles()
+		} else if req.EnableAlerting {
+			// Regenerates the rules files in case of enabling
+			s.rulesService.WriteVMAlertRulesFiles()
+		}
+
```

Comment:
Let's do this things only when alerts state actually changed (UI always sends full set of setting)

Similar changes: https://github.com/percona/pmm-managed/pull/617/files#diff-a990b29ab7e542cbefea97ccf6055f35a8f1486f1af794294dc542bf9e0e0b25R611

Reply:
Done